src/main/java/uk/gov/hmcts/divorce/solicitor/event/Applicant1SolicitorSwitchToSoleFo.java

```diff
@@ -0,0 +1,103 @@
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.model.UserRole;
+
+import static java.util.Collections.singletonList;
+import static uk.gov.hmcts.ccd.sdk.type.YesOrNo.YES;
+import static uk.gov.hmcts.divorce.common.event.ApplyForFinalOrder.APPLY_FOR_FINAL_ORDER;
+import static uk.gov.hmcts.divorce.divorcecase.model.ApplicationType.SOLE_APPLICATION;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingJointFinalOrder;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.FinalOrderRequested;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.APPLICANT_1_SOLICITOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.APPLICANT_2_SOLICITOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASE_WORKER;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
+
+@Slf4j
+@Component
+public class Applicant1SolicitorSwitchToSoleFo implements CCDConfig<CaseData, State, UserRole> {
+
+    public static final String APPLICANT_1_SOLICITOR_SWITCH_TO_SOLE_FO = "app1-sol-switch-to-sole-fo";
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+
+        new PageBuilder(configBuilder
+            .event(APPLICANT_1_SOLICITOR_SWITCH_TO_SOLE_FO)
+            .forStateTransition(AwaitingJointFinalOrder, FinalOrderRequested)
+            .showCondition("applicationType=\"jointApplication\""
+                + " AND dateApplicant1DeclaredIntentionToSwitchToSoleFo=\"*\""
+                + " AND finalOrderApplicant1NotifiedCanSwitchToSoleAfterIntention=\"Yes\"")
+            .name("Switched to sole final order")
+            .description("Switched to sole final order")
+            .grant(CREATE_READ_UPDATE, CASE_WORKER, SUPER_USER, APPLICANT_1_SOLICITOR)
+            .grantHistoryOnly(LEGAL_ADVISOR, APPLICANT_2_SOLICITOR)
+            .showSummary()
+            .showEventNotes()
+            .aboutToStartCallback(this::aboutToStart)
+            .aboutToSubmitCallback(this::aboutToSubmit))
+            .page("App1SolSwitchToSoleFO", this::midEvent)
+            .pageLabel(APPLY_FOR_FINAL_ORDER)
+            .complex(CaseData::getFinalOrder)
+            .mandatory(FinalOrder::getDoesApplicant1WantToApplyForFinalOrder)
+            .done();
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseData, State> details,
+                                                                  CaseDetails<CaseData, State> detailsBefore) {
+
+        log.info("Mid-event callback triggered for App1SolSwitchToSoleFO");
+
+        final CaseData data = details.getData();
+
+        if (!data.getFinalOrder().getDoesApplicant1WantToApplyForFinalOrder().toBoolean()) {
```

Comment:
any chance of doesApplicant1WantToApplyForFinalOrder is null at this point? just wondering if it can throw null pointer exception on toBoolean()?

Reply:
It shouldn't be as it is a mandatory field but I will change the condition anyway to be safe